x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/service/RestCreateServiceAccountTokenAction.java

```diff
@@ -48,7 +50,7 @@ public String getName() {
     protected RestChannelConsumer innerPrepareRequest(RestRequest request, NodeClient client) throws IOException {
         String tokenName = request.param("name");
         if (Strings.isNullOrEmpty(tokenName)) {
-            tokenName = UUIDs.base64UUID();
+            tokenName = new String(Hasher.SHA256.hash(new SecureString(UUIDs.base64UUID().toCharArray())));
```

Comment:
This feels like a weird fix.

We were using a random UUID in URL safe base64. But because it's possible for that to generate a string with a leading `_` (I think that's the problem), we're generating the Hash of that string and hex-encoding it.

The Hex encoding is the bit that actually solves the problem, but we're applying a bunch of other stuff as well.

Personally, I'd prefer that we drop leading `_` or regenerate rather than apply hashing and hex encoding just to work around a tiny incompatibility with the bast64 encoding.

Reply:
I thought about regenerating or replacing `_` or prefixing a random char or hex encoding the base64 bytes. But at the end, I chose sha256 since it is does not reduce the entropy (though it is not that important). It is also the most convenient.

Reply:
OK, but in this case it feels like we could ditch the whole UUID, base64 & SHA256 and just generate a string of bytes to be hex encoded.
In the end, that's all we actually want, and we're layering a lot of other algorithms on top of it.

Reply:
The token name is part of the document ID. So the UUID was initially chosen to minimize the possibility of collision since it is what used by the indexing itself. For this reason, I'd personally keep UUID (and subsequently base64 for it to be URL friendly). But SHA256 can be ditched. Previously I was thinking prefixing a random char before the UUID but didn't want to handle the randomness manually. But now it occurs to me that the prefix can and probably better be a fixed string, say `service_token_` or simply `st_`, which make the name more meaningful while keeping its randomness. I can update the PR to take this approach if you are OK with it.

Reply:
Yes some reasonable prefix (I would say `token_`, but I'm not too fussed) is a good choice

Reply:
Updated to prefix the name with `token_`. Thanks for the discussion!